scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1516,6 +1536,152 @@
                    max-considerable)
                  next-considerable))))))
 
+(defn job->task-txn
+  "Generate task transaction for a job."
+  [job metadata compute-cluster]
+  (let [{:keys [job/last-waiting-start-time job/uuid]} job
+        {:keys [executor task-id]} metadata
+        job-ref [:job/uuid uuid]
+        instance-start-time (now)]
+    [[:job/allowed-to-start? job-ref]
+     ; The job will remain waiting until the instance is running.
+     [:db/add job-ref :job/state :job.state/waiting]
+     (cond->
+      {:db/id (d/tempid :db.part/user)
+       :job/_instance job-ref
+       :instance/executor executor
+       :instance/executor-id task-id ; NB command executor uses the task-id as the executor-id 
+       :instance/hostname "Unknown"
+       :instance/ports 9876
+       :instance/preempted? false
+       :instance/progress 0
+       :instance/slave-id "Unknown"
+       :instance/start-time instance-start-time
+       :instance/status :instance.status/unknown
+       :instance/task-id task-id
+       :instance/compute-cluster (cc/db-id compute-cluster)}
+       last-waiting-start-time
+       (assoc :instance/queue-time
+              (- (.getTime instance-start-time)
+                 (.getTime ^Date last-waiting-start-time))))]))
+
+(defn jobs->kubernetes-task-metadata
+  "Generate sequence of task-metadata for jobs in a compute-cluster, suitable for Kubernetes Scheduler."
+  [jobs pool-name mesos-run-as-user compute-cluster]
+  (let [jobs->task-id (plumbing.core/map-from-keys (fn [_] (str (d/squuid))) jobs)]
+    (for [{:keys [job/name job/user job/uuid job/environment] :as job} jobs]
+      (let [pool-specific-resources
+            ((adjust-job-resources-for-pool-fn pool-name) job (tools/job-ent->resources job))]
+        (merge
+         (task/job->task-metadata compute-cluster mesos-run-as-user
+                                  job (jobs->task-id job))
+         {; NB: see `kubernetes/compute_cluster/autoscale!`
+          ; for how launching a real job pod without a hostname
+          ; compares to launching a synthetic pod.
+
+          ; Job constraints need to be expressed on a real job
+          ; pod so that if it triggers the cluster autoscaler,
+          ; CA will spin up nodes that will end up satisfying it. 
+          :pod-constraints (constraints/job->constraints job)
+
+          ; We need to label the pods handled completely by Kubernetes
+          ; so that we can treat them differently throughout the scheduler.
+          :pod-labels {"twosigma.com/scheduler" "kubernetes"}
```

Comment:
I'd consider a `cook` prefix here instead of just `twosigma.com`. Also, can this be in a `def`?

Reply:
Added the prefix. 

For now, I don't want to make this a def because it is accessed both here and in `cook.kubernetes.api`, which cannot be imported due to a cycle. This speaks to the eventual TODO here to move this task-metadata generation into probably `cook.kubernetes.task`. This would require refactoring `task/job->task-metadata`. I may be able to do that refactoring in this PR, but I need your help understanding some of the nuance there. I'm leaning towards adding this to list of cleanup tasks for post production experiment.